arch/arm/core/aarch32/fatal.c

```diff
@@ -104,13 +107,14 @@ void z_do_kernel_oops(const z_arch_esf_t *esf)
 	z_arm_fatal_error(reason, esf);
 #else
 	/* extra exception info is not collected for kernel oops
-	 * path today so we make a copy of the ESF and zero out
-	 * that information
+	 * by hardware, but callee regs are saved into a global
+	 * structure on ARMv7/ARMv8 by ARCH_EXCEPT()
 	 */
 	z_arch_esf_t esf_copy;
 
 	memcpy(&esf_copy, esf, offsetof(z_arch_esf_t, extra_info));
 	esf_copy.extra_info = (struct __extra_esf_info) { 0 };
+	esf_copy.extra_info.callee = &z_callee_saved;
```

Comment:
Should limit this to `CONFIG_ARMV7_M_ARMV8_M_MAINLINE`; otherwise, we will be seeing garbage callee saved register values for the rest of the ARM architecture variants.